network/manager.go

```diff
@@ -698,3 +687,54 @@ func (nm *networkManager) GetEndpointID(containerID, ifName string) string {
 	}
 	return containerID + "-" + ifName
 }
+
+func cnsEndpointInfotoCNIEpInfo(endpointInfo restserver.EndpointInfo, endpointID string) *EndpointInfo {
+	epInfo := &EndpointInfo{
+		Id:                  endpointID,
+		IfIndex:             EndpointIfIndex, // Azure CNI supports only one interface
+		ContainerID:         endpointID,
+		PODName:             endpointInfo.PodName,
+		PODNameSpace:        endpointInfo.PodNamespace,
+		NetworkContainerID:  endpointID,
+		SecondaryInterfaces: make(map[string]*InterfaceInfo),
+	}
+	// filling out the InfraNIC from the state
+	for ifName, ipInfo := range endpointInfo.IfnameToIPMap {
+		if ifName == InfraInterfaceName {
+			epInfo.IPAddresses = ipInfo.IPv4
+			epInfo.IPAddresses = append(epInfo.IPAddresses, ipInfo.IPv6...)
+			epInfo.IfName = ifName
+			epInfo.HostIfName = ipInfo.HostVethName
+			epInfo.HNSEndpointID = ipInfo.HnsEndpointID
+		} else { // filling out the SecondaryNICs from the state
+			interfaceInfo := &InterfaceInfo{
+				Name:      ifName,
+				IPConfigs: generateIPConfigfromState(ipInfo),
+				NICType:   ipInfo.NICType,
+			}
+			epInfo.SecondaryInterfaces[ifName] = interfaceInfo
+		}
+	}
+	return epInfo
+}
+
+func generateIPConfigfromState(ipInfo *restserver.IPInfo) []*IPConfig {
+	ipConfigs := []*IPConfig{}
+	for _, ip := range ipInfo.IPv4 {
+		ipConfig := &IPConfig{
+			Address: net.IPNet{
+				IP: ip.IP,
+			},
+		}
+		ipConfigs = append(ipConfigs, ipConfig)
+	}
+	for _, ip := range ipInfo.IPv6 {
+		ipConfig := &IPConfig{
+			Address: net.IPNet{
+				IP: ip.IP,
+			},
+		}
+		ipConfigs = append(ipConfigs, ipConfig)
+	}
+	return ipConfigs
```

Comment:
I thought our state files had a single struct that held both IPv4 and IPv6. Is it just that `epInfo` will only contain 2 `IPAddresses`?

Reply:
CNS IPInfo holds v4 and v6 separately.